src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,141 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import lombok.Value;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees
+                .stream()
+                .map(Employee::getPerson)
+                .collect(Collectors.averagingInt(Person::getAge));
+
+        assertEquals(33.66, expected, 0.1);
+    }
+
+    @Test
+    public void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        @Value
+        class PersonIntegerPair {
+            Person person;
+            Integer integer;
+        }
+
+        Person expected = employees
+                .stream()
+                .map(Employee::getPerson)
+                .map(person -> new PersonIntegerPair(person, person.getFullName().length()))
+                .max(Comparator.comparingInt(PersonIntegerPair::getInteger))
+                .map(PersonIntegerPair::getPerson)
+                .orElseThrow(RuntimeException::new);
+
+        assertEquals(expected, employees.get(1).getPerson());
+    }
+
+    @Test
+    public void findEmployeeWithMaximumDurationAtOnePosition() {
+        List<Employee> employees = getEmployees();
+
+        @Value
+        class PersonJobHistoryEntryPair {
+            Employee employee;
+            JobHistoryEntry jobHistoryEntry;
+        }
+
+        Employee expected = employees
+                .stream()
+                .flatMap(employee
+                        -> employee
```

Comment:
As I know, there is no code style where ```->``` is moved to the start of a new line.
Usually write like this:
```Java
employee -> 
        employee
````